graph-builder/src/graph.rs

```diff
@@ -142,5 +150,101 @@ fn create_graph(
             })
         })?;
 
+    graph.prune_abstract();
+
     Ok(graph)
 }
+
+/// Removes and retrieves the metadata value of the given `key` which is prefixed
+/// by `filter`
+fn remove_prefixed_label(
+    filter: &str,
+    release: &mut registry::Release,
+    key: &str,
+) -> Option<String> {
+    release
+        .metadata
+        .metadata
+        .remove(&format!("{}.{}", filter, key))
+}
+
+/// Remove releases which are labeled for removal.
+///
+/// The labels are assumed to have the syntax `{filter}.release-remove=<bool>`
+fn process_release_remove_label(
+    filter: &str,
+    release: registry::Release,
+) -> Option<registry::Release> {
+    let mut release = release;
+
+    if let Some(s) = remove_prefixed_label(filter, &mut release, "release.remove") {
+        // Filter releases which have "{prefix}.release-remove=true"
+        let remove = std::str::FromStr::from_str(&s).unwrap_or_else(|e| {
+            debug!("could not parse '{}': {}", s, e);
+            true
+        });
+
+        if remove {
+            debug!("Removing release '{:?}'", release);
```

Comment:
This sounds like a `trace` (unless you were planning to watch the labels live initially).

Reply:
That was the idea yes. Do you think it gets too verbose?

Reply:
I think initially it would be fine to double-check the behavior; later we can probably lower this if it becomes too verbose.